gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -123,29 +123,29 @@ private Reference getAgentReference(List<RCMRMT030101UK04Participant> participan
 
     private Optional<String> getParticipantReference(List<RCMRMT030101UK04Participant> participantList, String typeCode) {
         return participantList.stream()
-                .filter(participant -> hasTypeCode(participant, typeCode))
-                .filter(participant -> hasAgentReference(participant))
-                .map(RCMRMT030101UK04Participant::getAgentRef)
-                .map(RCMRMT030101UK04AgentRef::getId)
-                .map(II::getRoot)
-                .findFirst();
+            .filter(participant -> hasTypeCode(participant, typeCode))
+            .filter(this::hasAgentReference)
+            .map(RCMRMT030101UK04Participant::getAgentRef)
+            .map(RCMRMT030101UK04AgentRef::getId)
+            .map(II::getRoot)
+            .findFirst();
     }
 
     private Optional<String> getParticipant2Reference(RCMRMT030101UK04EhrComposition ehrComposition) {
         return ehrComposition.getParticipant2().stream()
-                .filter(participant2 -> participant2.getNullFlavor() == null)
-                .map(RCMRMT030101UK04Participant2::getAgentRef)
-                .map(RCMRMT030101UK04AgentRef::getId)
-                .map(II::getRoot)
-                .findFirst();
+            .filter(participant2 -> participant2.getNullFlavor() == null)
+            .map(RCMRMT030101UK04Participant2::getAgentRef)
+            .map(RCMRMT030101UK04AgentRef::getId)
+            .map(II::getRoot)
+            .findFirst();
     }
 
     private boolean hasAgentReference(RCMRMT030101UK04Participant participant) {
         return participant.getAgentRef() != null && participant.getAgentRef().getId() != null;
     }
 
     private boolean hasTypeCode(RCMRMT030101UK04Participant participant, String typeCode) {
-        return participant.getTypeCode().get(0).equals(typeCode);
+        return participant.getTypeCode() != null && participant.getTypeCode().get(0).equals(typeCode);
```

Comment:
If `typeCode` is null, then `participant.getTypeCode()` method returns an empty list (I've checked what this method does), so you need to check if `participant.getTypeCode()` has any elements, not if it's null (because it never is)